mne/io/curry/curry.py

```diff
@@ -0,0 +1,238 @@
+
+import os
+import re
+import numpy as np
+import mne
+
+from ..base import BaseRaw, _check_update_montage
+from ..meas_info import create_info
+from ..utils import _read_segments_file, warn, _find_channels, _create_chs
+
+
+def _read_curry_events(fname_base, curry_vers):
+    """
+    Read curry files and convert the data for mne.
+    Inspired by Matt Pontifex' EEGlab loadcurry() extension.
+
+    :param full_fname:
+    :return:
+    """
+
+    #####################################
+
+    EVENT_FILE_EXTENSION = {7: ['.cef', '.ceo'], 8: ['.cdt.cef', '.cdt.ceo']}
+
+    event_file = None
+    for ext in EVENT_FILE_EXTENSION[curry_vers]:
+        if os.path.isfile(fname_base + ext):
+            event_file = fname_base + ext
+
+    if event_file is not None:
+        curry_events = []
+        save_events = False
+        with open(event_file) as fid:
+            for line in fid:
+
+                if "NUMBER_LIST END_LIST" in line:
+                    save_events = False
+
+                if save_events:
+                    curry_events.append(line.split("\t"))
+
+                if "NUMBER_LIST START_LIST" in line:
+                    save_events = True
+
+        curry_events = np.array(curry_events, dtype=int)
+
+    else:
+        curry_events = None
+
+    # TODO: This returns events in a curry specific format. This might be reformatted to fit mne
+    return curry_events
+
+
+def _get_curry_version(file_extension):
+    """check out the curry file version"""
+
+    if 'cdt' in file_extension:
+        curry_vers = 8
+
+    else:
+        curry_vers = 7
+
+    return curry_vers
+
+
+def _check_missing_files(full_fname, fname_base, curry_vers):
+    """
+    Check if all neccessary files exist and return the path without extension
+     and its CURRY version
+     """
+
+    if curry_vers == 8:
+        for check_ext in [".cdt", ".cdt.dpa"]:
+            if not os.path.isfile(fname_base + check_ext):
+                raise FileNotFoundError("The following required file cannot be"
+                                        " found: %s. Please make sure it is "
+                                        "located in the same directory as %s."
+                                        % (fname_base + check_ext, full_fname))
+
+    if curry_vers == 7:
+        for check_ext in [".dap", ".dat", ".rs3"]:
+            if not os.path.isfile(fname_base + check_ext):
+                raise FileNotFoundError("The following required file cannot be"
+                                        " found: %s. Please make sure it is "
+                                        "located in the same directory as %s."
+                                        % (fname_base + check_ext, full_fname))
+
+
+def _read_curry_info(fname_base, curry_vers):
+
+    #####################################
+    # read parameters from the param file
+
+    CAL = 1e-6
+    INFO_FILE_EXTENSION = {7: '.dap', 8: '.cdt.dpa'}
+    LABEL_FILE_EXTENSION = {7: '.rs3', 8: '.cdt.dpa'}
```

Comment:
all those constants should live outside of the scope of the functions. 

They represent constants of the file, so when I read the file I get them all in the same place at the top, and if I've to change them they are all in the same place.

I understand the thought of having them close from where they are used is positive (especially if I'm reading the code in paper, or browsing from my phone), but as soon as the function grows a bit (or I need them in two different places) this advantage gets lost. And most editors allow you for quick jumps.